packages/dx-react-demos/src/material-ui/filtering/custom-filter-row.jsx

```diff
@@ -16,81 +13,73 @@ import {
   TableHeaderRow,
   TableFilterRow,
 } from '@devexpress/dx-react-grid-material-ui';
+
 import {
   generateRows,
+  globalSalesValues,
 } from '../../demo-data/generator';
 
-const toLowerCase = value => String(value).toLowerCase();
-const predicate = (value, filter) => {
-  if (filter.columnName === 'sex') {
-    return toLowerCase(value) === toLowerCase(filter.value);
-  }
-  return toLowerCase(value).indexOf(toLowerCase(filter.value)) > -1;
-};
-
 const styles = theme => ({
   cell: {
+    verticalAlign: 'top',
     width: '100%',
+    paddingTop: theme.spacing.unit + 4,
     paddingLeft: theme.spacing.unit,
     paddingRight: theme.spacing.unit,
-    paddingTop: theme.spacing.unit * 1.5,
-    verticalAlign: 'top',
   },
-  selectPlaceholder: {
-    color: theme.palette.text.hint,
-  },
-  inputRoot: {
+  input: {
     width: '100%',
   },
 });
 
-const emptyValue = ' ';
-
-const SexFilterCellBase = ({ setFilter, filter, classes }) => (
+const UnitsFilterCellBase = ({ filter, setFilter, classes }) => (
   <TableCell className={classes.cell}>
-    <Select
-      classes={{ select: filter ? null : classes.selectPlaceholder }}
-      value={filter ? filter.value : emptyValue}
-      onChange={event =>
-        setFilter(event.target.value !== emptyValue ? { value: event.target.value } : null)
-      }
-      input={<Input classes={{ root: classes.inputRoot }} />}
-      renderValue={value => (value === emptyValue ? 'Filter...' : value)}
-    >
-      <MenuItem value={emptyValue}>Any</MenuItem>
-      <MenuItem value={'Male'}>Male</MenuItem>
-      <MenuItem value={'Female'}>Female</MenuItem>
-    </Select>
+    <Input
+      className={classes.input}
+      type="number"
+      value={filter ? filter.value : ''}
+      onChange={e => setFilter(e.target.value ? { value: e.target.value } : null)}
+      placeholder="Filter..."
+      inputProps={{
+        style: { textAlign: 'right' },
+        min: 1,
+        max: 4,
+      }}
+    />
   </TableCell>
 );
 
-SexFilterCellBase.propTypes = {
+UnitsFilterCellBase.propTypes = {
+  filter: PropTypes.shape({
+    value: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]).isRequired,
+  }),
   setFilter: PropTypes.func.isRequired,
   classes: PropTypes.object.isRequired,
-  filter: PropTypes.object,
 };
 
-SexFilterCellBase.defaultProps = {
+UnitsFilterCellBase.defaultProps = {
   filter: null,
 };
 
-const SexFilterCell = withStyles(styles, { name: 'SexFilterCell' })(SexFilterCellBase);
+const UnitsFilterCell = withStyles(styles, { name: 'SexFilterCell' })(UnitsFilterCellBase);
```

Comment:
missed `{ name: 'SexFilterCell' }`, Looks so out of place in official [demo documentation](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/)

Reply:
@YashZolmajdi thank you for pointing that. The related PR #1750